github3/repos/release.py

```diff
@@ -46,15 +46,46 @@ def _update_attributes(self, release):
         self.published_at = self._strptime(release.get('published_at'))
         #: Name of the tag
         self.tag_name = release.get('tag_name')
+        #: URL to download a tarball of the release
+        self.tarball_url = release.get('tarball_url')
         #: "Commit" that this release targets
         self.target_commitish = release.get('target_commitish')
         upload_url = release.get('upload_url')
         #: URITemplate to upload an asset with
         self.upload_urlt = URITemplate(upload_url) if upload_url else None
+        #: URL to download a zipball of the release
+        self.zipball_url = release.get('zipball_url')
 
     def _repr(self):
         return '<Release [{0}]>'.format(self.name)
 
+    def archive(self, format, path=''):
+        """Get the tarball or zipball archive for this release.
+
+        :param str format: (required), accepted values: ('tarball',
+            'zipball')
+        :param path: (optional), path where the file should be saved
+            to, default is the filename provided in the headers and will be
+            written in the current directory.
+            it can take a file-like object as well
+        :type path: str, file
+        :returns: bool -- True if successful, False otherwise
+
+        """
+        resp = None
+        if format == 'tarball':
+            resp = self._get(self.tarball_url, allow_redirects=True,
```

Comment:
> Does GitHub sometimes respond with missing fields?

Any time you iterate over a resource, the responses are truncated severely. The values included change over time (sometimes more are included by default, other times they remove fields). I find it best not to rely on the API given how unstable it is for a "stable" API.

> Furthermore, what's the general philosophy around URLs in github3.py?

The URL for the resource itself will always be there so always use that. Build other things from scratch because the URL structure is one of the few things the API team doesn't change.

I had grand hopes of being able to use the URLs and URITemplates returned by the API, but those hopes were dashed. I leave those there for people to take advantage of the templates natively.